### PR TITLE
Fixed supplemental workflow experience

### PR DESCRIPTION
**Pull request by jgeurten**
Fixed the following bugs on the supplemental workflow - 

#137 - Audit mode is now not a valid option on the rule-options page
#138 - the Wizard hides the recommended blocklist checkboxes and will not merge these blocklists into supplemental policies
#139 - the Wizard will not allow the creation of deny policies since the WDAC engine will ignore deny rules in supplementals
#141 - the Wizard will now set the specified policy name in the supplemental xml

Split out base, supplemental, edit and merge workflows into 2 structs; 1 for tracking the policy being generated base vs supplemental and 1 tracking the workflow {new, edit, merge}. Updated the Wizard to reference the proper structure

### CODE REVIEW

**Review by jsuther1974 — APPROVED** [review]

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs [hunk 1]
```diff
@@ -623,10 +572,78 @@ private bool ReadSetRules(object sender, EventArgs e)
...
+            else
+            {
+                if (this.Policy._Format == WDAC_Policy.Format.MultiPolicy)
+                {
+                    // Multi-policy Format Policy Templates
+                    switch (this.Policy._PolicyTemplate)
+                    {
+                        case WDAC_Policy.NewPolicyTemplate.WindowsWorks:
+                            // Windows Works Mode 
+                            xmlPathToRead = System.IO.Path.Combine(this._MainWindow.ExeFolderPath, Properties.Resources.WindowsTemplate);// "DefaultWindows_Audit.xml");
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:607]: Now not necessary

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs [hunk 2]
```diff
@@ -623,10 +572,78 @@ private bool ReadSetRules(object sender, EventArgs e)
...
+                            break;
+
+                        case WDAC_Policy.NewPolicyTemplate.SignedReputable:
+                            // Signed and Reputable Mode
+                            xmlPathToRead = System.IO.Path.Combine(this._MainWindow.ExeFolderPath, Properties.Resources.SACTemplate); //"SignedReputable.xml");
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:612]: not necessary

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs [hunk 3]
```diff
@@ -623,10 +572,78 @@ private bool ReadSetRules(object sender, EventArgs e)
...
+                            break;
+
+                        case WDAC_Policy.NewPolicyTemplate.AllowMicrosoft:
+                            // Allow Microsoft mode
+                            xmlPathToRead = System.IO.Path.Combine(this._MainWindow.ExeFolderPath, Properties.Resources.MicrosoftTemplate);// "AllowMicrosoft.xml");
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:617]: not necessary

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs [hunk 4]
```diff
@@ -623,10 +572,78 @@ private bool ReadSetRules(object sender, EventArgs e)
...
+                    }
+                }
+                else
+                {
+                    // Legacy/single-policy Format Policy Templates
+                    switch (this.Policy._PolicyTemplate)
+                    {
+                        case WDAC_Policy.NewPolicyTemplate.WindowsWorks:
+                            // Windows Works Mode 
+                            xmlPathToRead = System.IO.Path.Combine(this._MainWindow.ExeFolderPath, Properties.Resources.WindowsSingleTemplate); //"DefaultWindows_Audit - SingleFormat.xml");
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:628]: not necessary

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs [hunk 5]
```diff
@@ -623,10 +572,78 @@ private bool ReadSetRules(object sender, EventArgs e)
...
+                            break;
+
+                        case WDAC_Policy.NewPolicyTemplate.SignedReputable:
+                            // Signed and Reputable Mode
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:632]: Careful of calling this a mode

Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:632]: so very many places

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs (hunk 4 above)
Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:628]: so very many places

File: WDAC-Policy-Wizard/app/src/Policy.cs [hunk 6]
```diff
@@ -98,6 +108,7 @@ public WDAC_Policy()
             this._PolicyTemplate = NewPolicyTemplate.None;
             this._PolicyType = PolicyType.None;
```
Comment (bowerscd-corp) [inline WDAC-Policy-Wizard/app/src/Policy.cs:109]: These should be default-initialized to 0, no?

Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/Policy.cs:109]: aka None

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs (hunk 1 above)
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:607]: This is still necessary. Maybe I am missing something?

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs [hunk 7]
```diff
@@ -623,10 +572,78 @@ private bool ReadSetRules(object sender, EventArgs e)
...
+                            break;
+                    }
+                }
+                else
+                {
+                    // Legacy/single-policy Format Policy Templates
+                    switch (this.Policy._PolicyTemplate)
+                    {
+                        case WDAC_Policy.NewPolicyTemplate.WindowsWorks:
+                            // Windows Works Mode
```
[the remark this answers is not in the document]
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:627]: We call the templates "modes". E.g. "WindowsWorks Mode"

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs (hunk 1 above)
Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:607]: I was (and all these not necessary comments are) referring to the comment at the end.

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:627]: Which is both semantically and functionally different from S Mode and SE Mode. Hence the "careful of calling them modes"

File: WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs (hunk 1 above)
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:607]: Ah, got it. Just clean up the comment?

Comment (bowerscd-corp) [reply WDAC-Policy-Wizard/app/src/ConfigTemplate_Control.cs:607]: yes

File: WDAC-Policy-Wizard/app/src/Policy.cs (hunk 6 above)
Comment (jgeurten) [reply WDAC-Policy-Wizard/app/src/Policy.cs:109]: yeah, you're right. Good catch!

**Pull request merged**